blogs/ecs-ec2-crossvpc-with-tls/appmesh-baseline.yml

```diff
@@ -0,0 +1,937 @@
+                  - 'xray:PutTraceSegments'
+                  - 'xray:PutTelemetryRecords'
+                  - 'xray:GetSamplingRules'
+                  - 'xray:GetSamplingTargets'
+                  - 'xray:GertSamplingStatisticSumaries'
+                  - 'servicediscovery:RegisterInstance'
+                  - 'servicediscovery:DeregisterInstance'
+                  - 'route53:CreateHealthCheck'
+                  - 'route53:ChangeResourceRecordSets'
+                  - 'route53:UpdateHealthCheck'
+                Resource: '*'
+
+  InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Roles:
+        - !Ref EC2InstanceRole
+
+  EC2InstanceSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Access to the instance
+      VpcId: !Ref VPC
+      SecurityGroupIngress:
+          - CidrIp: !FindInMap ['SubnetConfig', 'VPC', 'CIDR']
+            IpProtocol: -1
+          - IpProtocol: tcp
+            CidrIp: 0.0.0.0/0
+            FromPort: 22
+            ToPort: 22
+
+  RubyEC2InstanceLaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
+    DependsOn: 
+      - KeyPair
+      - PublicRoute
+    Properties:
+      LaunchTemplateName: !Sub Ruby-EC2Instance-LaunchTemplate-${AWS::StackName}
+      LaunchTemplateData:
+        KeyName: !GetAtt KeyPair.Name
+        ImageId: !Ref LatestAmiId
+        InstanceType: t2.medium
+        InstanceInitiatedShutdownBehavior: terminate
+        IamInstanceProfile:
+          Arn: !GetAtt InstanceProfile.Arn
+        SecurityGroupIds: 
+          - !Ref EC2InstanceSecurityGroup
+        UserData:
+          Fn::Base64:
+            Fn::Join:
+              - "\n"
+              -
+                - |
+                  #!/bin/bash -ex
+                  exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+                  # Install required libs
+                  yum install -y git gcc gcc-c++ make readline-devel openssl-devel sqlite-devel gmp-devel jq
+                  
+                  # Install rbenv
+                  git clone git://github.com/rbenv/rbenv.git /tmp/.rbenv
+                  echo 'export PATH="/tmp/.rbenv/bin:/usr/local/bin:$PATH"' >> /tmp/.bashrc
+                  echo 'eval "$(rbenv init -)"' >> /tmp/.bashrc
+                  source /tmp/.bashrc
+                  # Install ruby-build
+                  git clone git://github.com/rbenv/ruby-build.git /tmp/ruby-build
+                  cd /tmp/ruby-build
+                  ./install.sh
+                  rbenv install 2.5.1 && rbenv global 2.5.1
+                  # Install rails and bundler
+                  gem install --force rails bundler
+                  gem update --system
+                  # Clone the repo and build the app
+                  export RUBY_ROOT=/tmp/ecsdemo-frontend
+                  git clone https://github.com/ffeijoo/ecsdemo-frontend.git /tmp/ecsdemo-frontend
```

Comment:
We'd generally prefer centralized ownership of a repository like this. If this repo is ever deleted then the demonstration will be broken. :(